drafts/web-editor-monaco-with-language-server-support.md

```diff
@@ -0,0 +1,971 @@
+
+If we look in to `editorDidMount` callback implementation you can see there are few functiona that are called in order to get the editor ready and connect to LS.
+
+```typescript
+const editorDidMount: EditorDidMount = (editor) => {
+    registerLanguage();
+    const model = createModel();
+    editor.setModel(model);
+    connectToLs();
+    editor.focus();
+};
+```
+Let's go through those called functions one by one.
+
+**registerLanguage()**: This function will perform a crucial configuration for the Monaco editor which is registering our Custom Language.
+```typescript
+export const registerLanguage = () => {
+    monaco.languages.register({
+        id: HELLO_LANG_ID,
+        aliases: [HELLO_LANG_ID],
+        extensions: [HELLO_LANG_EXTENSION]
+    });
+}
+```
+This will tell Monaco editor to not to rely on defined languages and treat all source as a custom language. If developer didn’t do this Monaco editor won’t send messages using Monaco Language Client.
+
+**createModel()**: This will create a new Monaco Editor model with a file URI. If we didn’t create this model Monaco will use the default model which uses a in memory file URI which will cause issues when comes to LS.
+```typescript
+export const createModel = (): monaco.editor.ITextModel => monaco.editor.createModel(
+    '',
+    HELLO_LANG_ID,
+    monaco.Uri.parse(
+        `file:///hello-${Math.random()}${HELLO_LANG_EXTENSION}`
+    )
+);
+```
+this is just a mock file URI that has been added but in case you have actual file URI please add that path. Our WebEditor is not implemented to actually handle files on the file system.
+
+Next set the created editor model to mounted editor as below.
+```typescript
+editor.setModel(model);
+```
+After preparing the Monaco Editor, the next step is to connect the Monaco Language Client to WebSocket. This is where we call the connectToLs().
+```typescript
+connectToLs();
+```
+That's it for the implementation
+
+### Step 5: Starting up the language server and web editor
+Well That’s it. Now it should be ready to go.
+
+You can build the Frontend using below command
+
+```shell
+npm run build
+```
+And you can run the Frontend using below command
+```shell
+npm run dev
+```
+to start the language server use below command
+
+```shell
+java -jar target/hellols-0.0.1-SNAPSHOT.jar
+```
+Here is the final look.
+
+![React Web Editor](https://cdn.hashnode.com/res/hashnode/image/upload/v1719713207751/GDUJ_Rq1a.jpeg?auto=format)
```

Comment:
Definitely a nice to have but does hashnode also support gifs?

Reply:
it should yes and it would also support a codesandbox, in case this would run there. Would be pretty cool to have a working example here with all the code